### PR TITLE
GameMode, DXVK_ASYNC, ESYNC and FSYNC support

### PR DESCRIPTION
**Pull request by Zi-SH**
Pull adds in GameMode, DXVK_ASYNC, and provisional ESYNC and FSYNC support. 

- New options added in the WINE tab.
- GameMode and FSYNC set only to show if the system supports them.
- ESYNC and FSYNC still have some oddities to work out, so their SettingEntry doesn't toggle them yet.
- Defaults should be consistent with current defaults. ESYNC is on by default, but specifically disabled in Runner.

### CODE REVIEW

File: src/XIVLauncher.Core/Components/SettingsPage/Tabs/SettingsTabWine.cs [hunk 1]
```diff
@@ -23,6 +23,17 @@ public SettingsTabWine()
                 CheckVisibility = () => startupTypeSetting.Value == WineStartupType.Command
             },
 
+            new SettingsEntry<bool>("Enable GameMode", "Enable launching with GameMode optimizations.", () => Program.Config.GameModeEnabled ?? true, b => Program.Config.GameModeEnabled = b)
+            {
+                CheckVisibility = () => File.Exists("/usr/lib/libgamemodeauto.so.0")
```
Comment (goaaats) [inline src/XIVLauncher.Core/Components/SettingsPage/Tabs/SettingsTabWine.cs:28]: Do you think it'd be better to use CheckValidity here, to let people know that they have to install this first?

Comment (Zi-SH) [reply src/XIVLauncher.Core/Components/SettingsPage/Tabs/SettingsTabWine.cs:28]: Not going to lie, forgot it existed. I do think it'd be a preferable option to hiding it all together so I'll check it in the morning.

**Comment by goaaats** [comment]
Did someone test this? Should we merge #915 first? @marzent

**Comment by marzent** [comment]
> Did someone test this? Should we merge #915 first? @marzent

It makes sense to merge https://github.com/goatcorp/FFXIVQuickLauncher/pull/915  first, then ESYNC and FSYNC toggles should also be functional

**Comment by goaaats** [comment]
It's a draft, just let me know when you get to finish it up and I'll have a go at it 👍

**Comment by marzent** [comment]
Should be good now

**Comment by Zi-SH** [comment]
Re-implemented all of the optimizations as they were previously, but utilizing the new CompatibilityTool. Unlocked ESync and FSync after testing multiple times.

**Review by goaaats — APPROVED** [review]

**Pull request merged**